solution/index.js

```diff
@@ -0,0 +1,318 @@
+}
+//this functions get Array of string(tasks) and create them on the DOM
+function addTasksListToTheDOM(ulElement , list){
+    if (list.length > 0){
+        let liElement = createElement("li", list[0], ["task"])
+        liElement.addEventListener('dblclick' , changeTaskHandler)
+        ulElement.append(liElement)
+        addTasksListToTheDOM(ulElement , list.slice(1))
+    }
+    
+}
+//this function gets tasks object and create the tasks on the DOM
+function addTasksFromLocalStorageToDOM (tasks){
+    addTasksListToTheDOM(getElementOfSection("ul", "section-to-do") , tasks["todo"])
+    addTasksListToTheDOM(getElementOfSection("ul", "section-in-progress") , tasks["in-progress"])
+    addTasksListToTheDOM(getElementOfSection("ul", "section-done") , tasks["done"])
+}
+// this function gets item and list and delete this item from the list(delete only the first apperance of this item)
+function removeFromList(item , list){
+    if(list.length === 0){
+        return []
+    }else if(item === list[0]){
+        return list.slice(1)
+    } else{
+        return [list[0]].concat(removeFromList(item , list.slice(1)))
+    }
+
+}
+//this function gets type of task and string of task and remove item from the local storage
+function removeTaskFromLocalStorage(taskType , stringOfTask){
+    let tasksFromLocalStorage = JSON.parse(localStorage.getItem("tasks"))
+    let tasksOnlyOnOneType = tasksFromLocalStorage[taskType]
+    tasksOnlyOnOneType = removeFromList(stringOfTask , tasksOnlyOnOneType)
+    tasksFromLocalStorage[taskType] = tasksOnlyOnOneType
+    localStorage.setItem("tasks", JSON.stringify(tasksFromLocalStorage))
+}
+//this function gets ul element and string of task and remove li element in the ul that have this content inside
+function removeTaskFromDOM(ulElement , stringOfTask){
+    for (let i = 0 ; i < ulElement.children.length ; i++){
+        const liElement = ulElement.children[i];
+        if (liElement.textContent === stringOfTask){
+            ulElement.removeChild(liElement)
+        }
+    }
+}
+//this function get string of task and  id of section and remove it from the section DOM and local storage
+function removeTask(stringOfTask , sectionId){
+    removeTaskFromLocalStorage(sectionIdToString(sectionId) , stringOfTask)
+    removeTaskFromDOM(getElementOfSection("ul", sectionId), stringOfTask)
+}
+//this function move task to a new section and remove it from the past section DOM and local storage 
+function changeSectionForTask(stringOfTask , pastSectionId , newSectionId ){
+    if(pastSectionId !== newSectionId){
+        addTask(stringOfTask , newSectionId)
+        removeTask(stringOfTask , pastSectionId)
+    }
+    
+}
+//this function is the handler function keydown. 
+function moveSectionHandler(event){
+    if (document.querySelector('li:hover')) {
+        if (event.altKey && event.keyCode > 48 && event.keyCode < 52) {
+            const stringLi= document.querySelector('li:hover').textContent
```

Comment:
Don't query the DOM multiple times if you can save the element in a variable and reuse it